packages/flutter/lib/src/material/banner_theme.dart

```diff
@@ -0,0 +1,135 @@
+    this.contentTextStyle,
+  });
+
+  /// Default value for [MaterialBanner.backgroundColor].
+  ///
+  /// If null, [MaterialBanner] uses [ThemeData.canvasColor].
+  final Color backgroundColor;
+
+  /// Used to configure the [DefaultTextStyle] for the [MaterialBanner.content]
+  /// widget.
+  ///
+  /// If null, defaults to [ThemeData.textTheme.body1].
+  final TextStyle contentTextStyle;
+
+  /// Creates a copy of this object with the given fields replaced with the
+  /// new values.
+  MaterialBannerThemeData copyWith({
+    Color backgroundColor,
+    TextStyle contentTextStyle,
+  }) {
+    return MaterialBannerThemeData(
+      backgroundColor: backgroundColor ?? this.backgroundColor,
+      contentTextStyle: contentTextStyle ?? this.contentTextStyle,
+    );
+  }
+
+  /// Linearly interpolate between two Card themes.
+  ///
+  /// The argument `t` must not be null.
+  ///
+  /// {@macro dart.ui.shadow.lerp}
+  static MaterialBannerThemeData lerp(MaterialBannerThemeData a, MaterialBannerThemeData b, double t) {
+    assert(t != null);
+    return MaterialBannerThemeData(
+      backgroundColor: Color.lerp(a?.backgroundColor, b?.backgroundColor, t),
+      contentTextStyle: TextStyle.lerp(a?.contentTextStyle, b?.contentTextStyle, t),
+    );
+  }
+
+  @override
+  int get hashCode {
+    return hashValues(
+      backgroundColor,
+      contentTextStyle,
+    );
+  }
+
+  @override
+  bool operator ==(dynamic other) {
+    if (identical(this, other))
+      return true;
+    if (other.runtimeType != runtimeType)
+      return false;
+    final MaterialBannerThemeData typedOther = other;
+    return typedOther.backgroundColor == backgroundColor
+        && typedOther.contentTextStyle == contentTextStyle;
+  }
+
+  @override
+  void debugFillProperties(DiagnosticPropertiesBuilder properties) {
+    super.debugFillProperties(properties);
+    properties.add(ColorProperty('backgroundColor', backgroundColor, defaultValue: null));
+    properties.add(DiagnosticsProperty<TextStyle>('contentTextStyle', contentTextStyle, defaultValue: null));
+  }
+}
+
+/// An inherited widget that defines the configuration for
+/// [MaterialBanner]s in this widget's subtree.
+///
+/// Values specified here are used for [MaterialBanner] properties that are not
+/// given an explicit non-null value.
+class MaterialBannerTheme extends InheritedWidget {
+  /// Creates a banner theme that controls the configurations for
+  /// [MaterialBanner]s in its widget subtree.
+  MaterialBannerTheme({
+    Key key,
+    Color backgroundColor,
+    TextStyle contentTextStyle,
+    Widget child,
+  }) : data = MaterialBannerThemeData(
+         backgroundColor: backgroundColor,
+         contentTextStyle: contentTextStyle,
+       ),
+       super(key: key, child: child);
+
+  /// The properties for descendant [MaterialBanner] widgets.
+  final MaterialBannerThemeData data;
+
+  /// The closest instance of this class' [data] value that encloses the given
```

Comment:
nit: first sentence of dart doc is it's own paragraph.

https://dart.dev/guides/language/effective-dart/documentation#do-separate-the-first-sentence-of-a-doc-comment-into-its-own-paragraph

Reply:
done